### PR TITLE
Cannot publish session with '#' in name #5241

### LINKED ISSUE #5241: Cannot publish session with '#' in name
A user has a session named `Peer Evaluation #1` which gives an 'unauthorized access' when trying to publish. Need a quick fix.

### PR DESCRIPTION
**Pull request by bamboo3250**
Fixes #5241
@wkurniawan07 @kanghj anyone available to review this soon?

### CODE REVIEW

File: src/main/java/teammates/common/util/StringHelper.java [hunk 1]
```diff
@@ -351,6 +351,21 @@ public static String recoverFromSanitizedText(String str) {
...
+    public static String recoverFromSanitizedForJsText(String str) {  
+        if (str == null) {
+            return null;
+        }
+        
+        return recoverFromSanitizedText(str)
+                  .replace("\\\\", "\\")
+                  .replace("\\\"", "\"")
+                  .replace("\\'", "'")
+                  .replace("\\#", "#");
```
Comment (kanghj) [inline src/main/java/teammates/common/util/StringHelper.java]: While this works, can we just remove the sanitizeForJs for session names? None of the characters other than # is allowed in session names.

Comment (bamboo3250) [inline src/main/java/teammates/common/util/StringHelper.java]: If there is no other special character allowed in session names besides `#` then that should be fine.

**Comment by bamboo3250** [comment]
@kanghj okay done. I have removed those sanitizedForJs() and appended a few `#` characters in session to improve the test.

**Comment by kanghj** [comment]
Ok, looks good to me

**Comment by wkurniawan07** [comment]
@damithc should I include this in V5.77 or create a new milestone V5.77.01?

**Comment by damithc** [comment]
I've merged it to release and tag as V5.77.01
To keep the record clean, we should create a separate milestone for V5.77.01

**Pull request merged**